src/gtkballs.c

```diff
@@ -16,12 +16,73 @@
 #include "prefs.h" /* preferences */
 #include "scoreboard.h" /* read_score, score_setup */
 #include "gfx.h"
-#include "child.h"
 #include "theme.h"
 #include "inputname.h" /* input_name_dialog */
 #include "game.h"
 #include "mainmenu.h"
 #include "mainwin.h"
+#include <sys/types.h>
+#include <sys/stat.h> /* mkdir */
+
+
+#define P_DIR "games"
+char * get_config_dir_file (const char * file)
+{
+   /* returns a path that must be freed with g_free */
+   char * config_home = NULL;
+   char * res = NULL;
+   int freestr = 0;
+#if __MINGW32__
+   config_home = getenv ("LOCALAPPDATA"); /* XP */
+   if (!config_home) {
+      config_home = getenv ("APPDATA");
+   }
+#else
+   const char * HOME = NULL;
+   config_home = getenv ("XDG_CONFIG_HOME");
+   if (!config_home) {
+      HOME = getenv ("HOME");
```

Comment:
Is it worth manually getting value of `$HOME`, when g_get_home_dir does the same?

https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-get-home-dir

Reply:
G_get_hom3_dir checks the HOME variable since glib 2.36 I think, which is an incompatible change, and the app compiles with gtk released in 2008, 2.20 perhaps. That was the reason

In fact I see he function can be heavily simplified, but the result would not be as consistent as it could be, need to read glibs source to see what it does in older versions.

Reply:
Ok, thank you. I've just checked the doc but it didn't occur to me that there would be such a big difference between versions.